src/Altinn.App.Core/Features/Payment/Providers/Nets/NetsPaymentProcessor.cs

```diff
@@ -0,0 +1,138 @@
+         * Entering the amount 100 corresponds to 1 unit of the currency entered, such as e.g. 1 NOK.
+         */
+        var payment = new NetsCreatePayment()
+        {
+            Order = new NetsOrder
+            {
+                Amount = (int)(orderDetails.TotalPriceIncVat * 100),
+                Currency = orderDetails.Currency,
+                Reference = orderDetails.OrderReference,
+                Items = orderDetails.OrderLines.Select(l => new NetsOrderItem()
+                {
+                    Reference = l.Id,
+                    Name = l.Name,
+                    Quantity = l.Quantity,
+                    Unit = l.Unit,
+
+                    UnitPrice = (int)(l.PriceExVat * 100),
+                    GrossTotalAmount = (int)(l.PriceExVat * 100 * l.Quantity * (1 + l.VatPercent / 100)),
+                    NetTotalAmount = (int)(l.PriceExVat * 100 * l.Quantity),
+                    TaxAmount = (int)(l.PriceExVat * 100 * l.Quantity * (l.VatPercent / 100)),
+                    TaxRate = (int)(l.VatPercent * 100),
+                }).ToList(),
+            },
+            MyReference = instance.Id.Split('/')[1],
+            Checkout = new NetsCheckout
+            {
+                IntegrationType = _settings.IntegrationType,
+                TermsUrl = _settings.TermsUrl,
+                ReturnUrl = altinnAppUrl,
+                CancelUrl = altinnAppUrl,
+                Appearance = new NetsApparence
+                {
+                    DisplayOptions = new NetsApparence.NetsDisplayOptions
+                    {
+                        ShowOrderSummary = _settings.ShowOrderSummary,
+                        ShowMerchantName = _settings.ShowMerchantName,
+                    },
+                },
+                Charge = true
+            },
+        };
+
+        HttpApiResult<NetsCreatePaymentSuccess> httpApiResult = await _netsClient.CreatePayment(payment);
+        if (!httpApiResult.IsSuccess || httpApiResult.Result?.HostedPaymentPageUrl is null)
+        {
+            throw new PaymentException("Failed to create payment\n" + httpApiResult.Status + " - " + httpApiResult.RawError);
+        }
+
+        string hostedPaymentPageUrl = httpApiResult.Result.HostedPaymentPageUrl;
+        string paymentId = httpApiResult.Result.PaymentId;
+
+        return new PaymentInformation
+        {
+            PaymentReference = paymentId,
+            RedirectUrl = hostedPaymentPageUrl,
+            OrderDetails = orderDetails,
+            Status = PaymentStatus.Created
+        };
+    }
+
+    /// <inheritdoc />
+    public async Task CancelPayment(Instance instance, string paymentReference)
+    {
+        await _netsClient.CancelPayment(paymentReference);
+    }
+
+    /// <inheritdoc />
+    public async Task<PaymentStatus?> GetPaymentStatus(Instance instance, string paymentReference, decimal expectedTotalIncVat)
```

Comment:
Any input on what kinds of verifications we should do to check if the payment is in good health? Right now I'm just doing a simple sum comparison. Charged amount == our local calculation of total inc VAT.